nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -212,9 +215,21 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
         final Map<String, String> statePropertyMap = new HashMap<>(stateMap.toMap());
 
         //If an initial max value for column(s) has been specified using properties, and this column is not in the state manager, sync them to the state property map
-        for(final Map.Entry<String,String> maxProp : maxValueProperties.entrySet()){
-            if (!statePropertyMap.containsKey(maxProp.getKey().toLowerCase())) {
-                statePropertyMap.put(maxProp.getKey().toLowerCase(), maxProp.getValue());
+        for (final Map.Entry<String, String> maxProp : maxValueProperties.entrySet()) {
+            String maxPropKey = maxProp.getKey().toLowerCase();
+            String fullyQualifiedMaxPropKey = getStateKey(tableName, maxPropKey);
+            if (!statePropertyMap.containsKey(fullyQualifiedMaxPropKey)) {
+                String newMaxPropValue;
+                // If we can't find the value at the fully-qualified key name, it is possible (under a previous scheme)
+                // the value has been stored under a key that is only the column name. Fall back to check the column name,
+                // but store the new initial max value under the fully-qualified key.
+                if (statePropertyMap.containsKey(maxPropKey)) {
+                    newMaxPropValue = statePropertyMap.get(maxPropKey);
+                } else {
+                    newMaxPropValue = maxProp.getValue();
```

Comment:
Agreed, I was faced with the choice of 1) an extra query for QueryDatabaseTable, which might happen when scheduled (if using static table names and max-value columns) or once for each table, or 2) not allowing incoming flow files for QDT.  I chose option 2 for the reason you point out, the other processors offer that functionality in a more robust manner IMHO. It sounds like we're on the same page, I will add @SeeAlso references to the processors you mention, along with some more text in the CapabilityDescription for QDT to refer to the other processors.